src/golang/lib/job/process.go

```diff
@@ -268,6 +270,15 @@ func (j *ProcessJobManager) Launch(
 		return ErrJobAlreadyExists
 	}
 
+	// For function operators, we need to set the directory path to expand the function into here.
+	var fnExtractStoragePath string
+	if spec.Type() == FunctionJobType {
+		if fnSpec, ok := spec.(*FunctionSpec); ok {
+			fnExtractStoragePath = path.Join(j.conf.OperatorStorageDir, uuid.New().String())
+			fnSpec.FunctionExtractPath = fnExtractStoragePath
+		}
+	}
+
```

Comment:
This is a bit annoying, because we need to set the field on the Command object at this level, but we also need to modify the spec if its a function. Ideally, the caller would pass in the spec with the extract field already populated, but the caller does not have access to `OperatorStorageDir` at the moment.